### PR TITLE
home screen: the gateway to tournaments

### PR DESCRIPTION
**Pull request by williambertrand**
### Home Screen
- This is totally up for being changed in the future, but just wanted to have an easy entry point into:
    - Viewing a user's current tournaments
    - Creating or joining a new one
- This only adds UI, no requests are made yet, will do so in another PR after backend is ready for fetching user tourny's 

### Notes

- `models.tsx` - I like to have a models file like this specifically with types relating to the data being passed around. This way it doesn't get all mixed in with the other types declared in the existing `types.tsx`. (Also, if we ever move to redux, this file will basically become the model store, although TBH I don't look forward to that day)
- Added `ScreenWithNavigation` for all those pesky instances when you have a screen that only needs a Navigation prop (it happens a lot)

| Home Empty | Home w/ tournaments |
| --- | --- |
| Screen Shot 2021-12-21 at 9 09 57 AM | Screen Shot 2021-12-21 at 9 06 49 AM |

Light:
| Home Empty | Home w/ tournaments |
| --- | --- |
| Screen Shot 2021-12-21 at 9 09 57 AM | Screen Shot 2021-12-21 at 9 06 49 AM |

### CODE REVIEW

File: components/Themed.tsx [hunk 1]
```diff
@@ -101,6 +102,16 @@ export function View(props: ViewProps) {
   return <DefaultView style={[{ backgroundColor }, style]} {...otherProps} />;
 }
 
+export function SafeAreaView(props: ViewProps) {
```
Comment (probicheaux) [inline components/Themed.tsx:105]: https://reactnative.dev/docs/safeareaview

Looks like this only applies to IOS. Is this still safe to use with android? Maybe I should just "build" it and try it out

File: components/Themed.tsx [hunk 2]
```diff
@@ -125,14 +136,19 @@ export function LinkButton(props: LinkButtonProps) {
...
         {title}
       </DefaultText>
     </DefaultTouchableOpacity>
 
   );
 }
 
+LinkButton.defaultProps = {
+  lightColor: '#2684FF',
+  darkColor: '#fff',
```
Comment (probicheaux) [inline components/Themed.tsx]: Should this go in/be sourced from `constants/Colors.ts`? I thiink the point of the lightColor/darkColor props is as an override to the default themed components

File: components/Tournaments/TournamentList.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import { ViewStyle, StyleSheet } from 'react-native';
+import { Tournament } from '../../models';
+import { FlatList, View, Text, LinkButton } from '../Themed'
+
+interface TournamentListProps {
+    navigation: NavigationProp<any>
+    tournaments: Tournament[]
+}
+
+const TournamentList = ({ tournaments, navigation }: TournamentListProps) => {
```
Comment (probicheaux) [inline components/Tournaments/TournamentList.tsx:12]: noice

File: models.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export interface Tournament {
```
Comment (probicheaux) [inline models.tsx:1]: dope, yeah,  I should be adding expected API response formats here, once those are more nailed down I guess. I think `date` will be a relevant field here eventually

File: screens/HomeScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        </View>
+        
+        <TournamentList
+            navigation={navigation}
+            tournaments={myTournaments} 
+        />
+
+        <TouchableOpacity
+            onPress={() => navigation.navigate('CreateTournament')}
+            style={[ButtonStyles.container, {width: '50%'}]}
```
Comment (probicheaux) [inline screens/HomeScreen.tsx:27]: TIL about this style join thingy in react native

**Review by probicheaux — APPROVED** [review]
All looks good to me! Only thing that is missing(?) is editing `navigation/LinkingConfiguration.ts` to reflect the new changes to the navigation structure (I think?), but to be honest I don't really know how that's supposed to work so... maybe I will read up on that more and make sure the whole project is conforming to the right guidelines

 * https://reactnavigation.org/docs/deep-linking
 * https://reactnavigation.org/docs/configuring-links

File: components/Themed.tsx (hunk 1 above)
Comment (williambertrand) [reply components/Themed.tsx:105]: Yeah my understanding is this is to help out with devices with "nothces" - i dont think androids have those?

**Comment by williambertrand** [comment]
> editing `navigation/LinkingConfiguration.ts` to reflect the new changes to the navigation structure

Good call, the way I've handled this before is just ignore this file until I finally add a deep link, and then make everything match as needed, but def better idea to keep it in sync while working on this lol

File: components/Themed.tsx (hunk 2 above)
Comment (williambertrand) [reply components/Themed.tsx]: Oh yeah, good catch - I didn't quite get that light/dark color props thing at first.  Moved it over.

**Pull request merged**